### PR TITLE
feat: add libmbus

### PR DESCRIPTION
**Pull request by gocarlos**
Specify library name and version:  **lib/1.0**

depends on https://github.com/rscada/libmbus/pull/151

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libmbus/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200323":
+    sha256: 6607832790794c6e7b92a3c768868781f6dfacd48f80953e1653a0d48d03fdfe
+    url: https://github.com/gocarlos/libmbus/archive/a967850d1689e8b439b6501529b584c2696d2b17.zip
```
Comment (madebr) [inline recipes/libmbus/all/conandata.yml]: Please use the upstream url: https://github.com/rscada/libmbus

I guess this is temporary to test [your cmake pull request](https://github.com/rscada/libmbus/pull/151)?

File: recipes/libmbus/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibMbusConan(ConanFile):
+    name = "libmbus"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/gocarlos/libmbus/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """Meter-bus library and utility programs"""
+    topics = ("mbus")
```
Comment (madebr) [inline recipes/libmbus/all/conanfile.py]:
```suggestion
    topics = "conan", "mbus", "metering", "iot", "meter", "bus", "protocol"
```

File: recipes/libmbus/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder,
+                                 "lib", "libmbus", "cmake"))
```
Comment (madebr) [inline recipes/libmbus/all/conanfile.py]: Is there something in `os.path.join(self.package_folder, "lib", "libmbus")`?

File: recipes/libmbus/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(test_package)
+
+# set(CMAKE_VERBOSE_MAKEFILE TRUE)
```
Comment (madebr) [inline recipes/libmbus/all/test_package/CMakeLists.txt]:
```suggestion
```

File: recipes/libmbus/all/conandata.yml (hunk 1 above)
Comment (gocarlos) [reply recipes/libmbus/all/conandata.yml]: Yes

**Comment by uilianries** [comment]
Hi @gocarlos is it still on WIP?

**Comment by gocarlos** [comment]
> Hi @gocarlos is it still on WIP?

yes, until https://github.com/rscada/libmbus/pull/151 is merged...

cc/ @lategoodbye

**Comment by gocarlos** [comment]
ready to review

File: recipes/libmbus/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os not in ["Linux"]:
```
Comment (madebr) [inline recipes/libmbus/all/conanfile.py:40]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        if self.settings.os not in ["Linux"]:
```

**Pull request merged**